Benchmarks/Benchmarks/JSON/JSONBenchmark.swift

```diff
@@ -0,0 +1,109 @@
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+import Benchmark
+import func Benchmark.blackHole
+
+#if os(macOS) && USE_PACKAGE
+
+// Import the package
+import FoundationEssentials
+
+// Also import the system Foundation, because we need it for `Bundle` to find resources
+import Foundation
+
+// Use the types from the package, to disambiguate from the system Foundation
+typealias _Data = FoundationEssentials.Data
+typealias _URL = FoundationEssentials.URL
+typealias _JSONEncoder = FoundationEssentials.JSONEncoder
+typealias _JSONDecoder = FoundationEssentials.JSONDecoder
+#else
+import Foundation
+
+// Use the types from Foundation
+typealias _Data = Foundation.Data
+typealias _URL = Foundation.URL
+typealias _JSONEncoder = Foundation.JSONEncoder
+typealias _JSONDecoder = Foundation.JSONDecoder
+#endif
+
+func path(forResource name: String) -> _URL? {
+#if FOUNDATION_FRAMEWORK
+    // This benchmark config puts resources in the main bundle
+    guard let url = Bundle.main.url(forResource: name, withExtension: nil) else { return nil }
+#else
+    // This is always package-based, and uses the module to find resources
+    guard let url = Bundle.module.url(forResource: name, withExtension: nil) else { return nil }
+#endif
+    return _URL(fileURLWithPath: url.path)
+}
+
+let benchmarks = {
+    Benchmark.defaultConfiguration.maxIterations = 1_000_000_000
+    Benchmark.defaultConfiguration.maxDuration = .seconds(3)
+    Benchmark.defaultConfiguration.scalingFactor = .kilo
+    Benchmark.defaultConfiguration.metrics = [.cpuTotal, .throughput]
+    
+    let canadaPath = path(forResource: "canada.json")
+    let canadaData = try! _Data(contentsOf: canadaPath!)
+    let canada = try! _JSONDecoder().decode(FeatureCollection.self, from: canadaData)
+
+    let twitterPath = path(forResource: "twitter.json")
+    let twitterData = try! _Data(contentsOf: twitterPath!)
+    let twitter = try! _JSONDecoder().decode(TwitterArchive.self, from: twitterData)
+
+    Benchmark("Canada-decodeFromJSON") { benchmark in
+        let result = try! _JSONDecoder().decode(FeatureCollection.self, from: canadaData)
```

Comment:
The closure passed to the Benchmark initializer can throw, so we can just use `try` here. (The benchmarks closure however cannot throw, sadly.)